source/adios2/engine/bp5/BP5Engine.h

```diff
@@ -172,6 +172,8 @@ class BP5Engine
     MACRO(StatsBlockSize, SizeBytes, size_t, DefaultStatsBlockSize)                                \
     MACRO(Threads, UInt, unsigned int, 0)                                                          \
     MACRO(UseOneTimeAttributes, Bool, bool, true)                                                  \
+    MACRO(UseSelectiveMetadataAggregation, Bool, bool, true)                                       \
+    MACRO(OneLevelGatherSizeLimit, Int, int, 6000)                                                 \
```

Comment:
`OneLevelGatherSizeLimit`: Size of what? To me, it first suggests a metadata size limit. Propose new names: `OneLevelGatherMPISizeLimit`, `OneLevelGatherNprocLimit`
Change could go into another PR with lines in the documentation ;-)

Reply:
I would actually call it `OneLevelGatherRanksLimit` since the threshold is given by the number of ranks that are part of the gather right?

Reply:
I like OneLevelGatherRanksLimit.  I've changed to that, eliminated a couple of unused (and undocumented) parameters and added some docs on the parameters introduced here.  Turns out I miscounted and UseSelectiveMetadataAggregation was longer than anything and I had to reformat the table a bit to get it to fit.